src/tracker-component.js

```diff
@@ -60,6 +60,41 @@ const track = <T>(userData : UserData, trackingType : TrackingType, trackingData
     });
 };
 
+const exit = () => { // returns true if modal was shown
+    // TODO: if identified - do nothing
+    const email = localStorage.getItem('paypal-cr-user');
+    if (email !== null) {
+        console.log('[exit] no email');
+        return false;
+    }
+    // TODO: does user have no items in cart - do nothing
+    const cart = localStorage.getItem('paypal-cr-cart')
+    if (!cart.items) {
+        console.log('[exit] no items');
+        return false;
+    }
+    console.log('cart:', cart)
+    // TODO: has this been shown in past 7 days - do nothing
+    const sevenDays = 1000 * 60 * 60 * 24 * 7;
+    const lastSeen = localStorage.getItem('paypal-cr-lastseen');
+    if (Date.now() - lastSeen < sevenDays) {
+        console.log('[exit] seen < 7 days ago');
+        return false;
+    }
+    // TODO: render experience
+    console.log('render experience....');
+    return true;
+}
+
+const start = () => {
+    document.body.addEventListener('mousemove', e => {
```

Comment:
maybe try to debounce and limit it to only being called once, maybe something like this

```
const debounce = (f, ms) => {
    let timeoutId;
    return (...args) => {
        clearTimeout(timeoutId);
        timeoutId = setTimeout(() => f(...args), ms);
    };
};

const limitCalls = (f, callLimit = 1) => {
    let calls = 0;
    return (...args) => {
        if (calls >= callLimit) {
            return;
        }
        calls += 1;
        return f(...args);
    };
};

// handleExit only gets called 1 time and is debounced for 500 ms
const exitEventHandler = debounce(limitCalls(handleExit, 1), 500);
document.body.addEventListener('mousemove', exitEventHandler);
```